### ISSUE TITLE
CtrlP Breaks with Dashes in Path

### ISSUE THREAD

**Issue by alexlafroscia** [ctrlpvim/ctrlp.vim#93]
I ran into an issue with CtrlP not being able to find files for a particular project, and after some searching online found [this thread](https://github.com/kien/ctrlp.vim/issues/566) on the original CtrlP repo.  As mentioned there, the issue seems to be that there are dashes in the path name – other people experienced that issue, and it seems to be the problem for me as well.

**Comment by mattn** [comment]
please give us how to reproduce.

**Comment by alexlafroscia** [comment]
If you try to use CtrlP in on a file when the CMD is something like `~/my-path` (with a hyphen in it) then no files are recognized, it just shows `== NO ENTRIES ==`.

**Comment by mattn** [comment]
I checked out the repository you pointed, but I can't repro.

![](http://go-gyazo.appspot.com/a7ece290c65b4e1f.png)

**Comment by alexlafroscia** [comment]
Hmm, that's interesting... This is what I'm seeing
![screen shot 2015-04-25 at 3 29 49 pm](https://cloud.githubusercontent.com/assets/1645881/7334423/f7746e12-eb5f-11e4-9643-2af255d5f506.png)

**Comment by mattn** [comment]
Could you please show me your configurations about ctrlp?

**Comment by alexlafroscia** [comment]
Sure.  These are enabled only if `ag` is present (and it is)

```viml
let g:ctrlp_user_command = 'ag %s -l --nocolor -g ""'
let g:ctrlp_use_caching = 0
```

These are enabled all the time

```viml
let g:ctrlp_custom_ignore = '\v[\/]\.(git|hg|svn)$'
let g:ctrlp_show_hidden = 1
```

**Comment by mattn** [comment]
I can't reproduce it with your configuration.

![](http://go-gyazo.appspot.com/7cca8c4f064c9f8e.png)

**Comment by alexlafroscia** [comment]
Alright, I'll see if I can play with it more and figure out exactly what about the repository is causing it.  Did you try to close that repo into a path that matches? IE: clone it into `~GOPATH/src/github.com/Pitt-CSC/icarus-backend`?

**Comment by mattn** [comment]
![](http://go-gyazo.appspot.com/0155e595c82c835f.png)
still not.

**Comment by alexlafroscia** [comment]
Alright, I was able to track down the issue to this configuration line

```viml
set grepprg=ag\ --nogroup\ --nocolor
```

**Comment by mattn** [comment]
unfortunately, still not. seems not depend on `grepprg` on my environment.

**Comment by mattn** [comment]
Are you put `~/.agignore`?

**Comment by alexlafroscia** [comment]
No, should I ?

**Comment by mattn** [comment]
so did you figure out that is wrong?

**Comment by alexlafroscia** [comment]
Nope, not any more than I told you, that particular directory with that particular config option is broken

**Comment by mattn** [comment]
Could you please check `:lcd` when you get "NO ENTRY"?

**Comment by alexlafroscia** [comment]
Alright, after getting "NO ENTRY", I dismissed Ctrl-P and it returned `/Users/alex`

**Comment by mattn** [comment]
current directory seems to be changed to home directory.

**Comment by brandoniffert** [comment]
@alexlafroscia Did you ever figure out what the problem was? I just ran into this with [this repo](https://github.com/roots/sage), it has `wp-content` in the path but my other Wordpress projects have never had this problem. `:lcd` returns my home directory and I don't have `grepprg` set.

**Comment by alexlafroscia** [comment]
Not really, unfortunately. I can't remember if I did something in particular but it eventually resolved itself.

**Comment by thenickcox** [comment]
Not sure if this is helpful, but I'm seeing this on my [local fork of the site for Elixir](https://github.com/thenickcox/elixir-lang.github.com), which has a hyphen in it. Like @alexlafroscia and @brandoniffert, `lcd` returns my home directory. Then when I switch to another iTerm tab where ctrlp was working, it then changes to my home directory, too. But there, instead of returning `== NO ENTRIES ==`, it returns the contents of my home directory.

**Comment by wsdjeg** [comment]
@alexlafroscia 
i get the same issue, and i fixed it by change this option
let g:ctrlp_match_func = { 'match': 'pymatcher#PyMatch' }
maybe you can have a look at my vimrc,
https://github.com/wsdjeg/DotFiles/blob/master/vimrc
line169-line260

**Comment by dansomething** [comment]
Are there any updates on this issue? It still seems to be a problem for me. I have same issue with an Angular 4 repository which uses hyphens in directories and file names. I am overriding ctrlp_user_command in my vimrc with

    let g:ctrlp_user_command = 'ag %s -l --nocolor --hidden -g ""'

When I disable that config I get results every time. The strange thing is that when I have it enabled I get "NO ENTRIES" about half the time. Retries usually end up loading results.

**Comment by alexlafroscia** [comment]
I don't use `ctrl-p` anymore, so no updates on my end :P

**Comment by mattn** [comment]
If you still have this problem, try to reproduce with another command. CtrlP doesn't have tricks especially for ag.
